sound/soc/intel/boards/skl_hda_dsp_generic.c

```diff
@@ -168,6 +168,28 @@ static int skl_hda_fill_card_info(struct snd_soc_acpi_mach_params *mach_params)
 	return 0;
```

Comment:
@jason77-wang For the commit message. The driver is already using autosuspend, so the summary is not really accurate. How about:

ASoC: intel/skl/hda - set autosuspend timeout for hda analog codecs

Reply:
the commit message is also a bit unclear. Assume that all devices are already suspended, and the jack is plugged. *something* causes a wake-up to handle the first unsolicited answer for headphone. If the  devices go back to suspend, what causes the second unsolicited answer for mic to be missed?
In other words, why can we handle the first case and not the second? It's not the suspended status, something is missing in the explanation.

Reply:
> @jason77-wang For the commit message. The driver is already using autosuspend, so the summary is not really accurate. How about:
> 
> ASoC: intel/skl/hda - set autosuspend timeout for hda analog codecs

OK, will change it.

Reply:
@plbossart This was the recommendation from Realtek. Jack detection causes the wake-up from D3, but the detection routine takes time to complete and sent out all the unsolicited response, if driver puts codec immediately back to sleep, it won't wake up again and we lose some responses -- i.e. this is a feature of some codecs. I think the commit msg accurately describes what is seen.

Reply:
> the commit message is also a bit unclear. Assume that all devices are already suspended, and the jack is plugged. _something_ causes a wake-up to handle the first unsolicited answer for headphone. If the devices go back to suspend, what causes the second unsolicited answer for mic to be missed?
> In other words, why can we handle the first case and not the second? It's not the suspended status, something is missing in the explanation.

That is sth I can't explain, I did a test before, I did not enable the unsol event on headphone pin, only enable the unsol event on the mic pin,  then there is no unsol interrupt anymore if it is in suspend state. That means if don't depend on the headphone, the mic can't trigger unsol event  if the codec is suspended.